system_tests/gas_dim_tx_op_call_test.go

```diff
@@ -0,0 +1,985 @@
+	defer cancel()
+	defer cleanup()
+
+	_, caller := deployGasDimensionTestContract(t, builder, auth, gas_dimensionsgen.DeployCaller)
+	calleeAddress := common.HexToAddress("0x00000000000000000000000000000000DeaDBeef")
+
+	// prefund callee address
+	_, _ = builder.L2.TransferBalanceTo(t, "Owner", calleeAddress, big.NewInt(1e17), builder.L2Info)
+
+	receipt := callOnContractWithOneArg(t, builder, auth, caller.ColdNoTransferMemExpansion, calleeAddress)
+
+	TxOpTraceAndCheck(t, ctx, builder, receipt)
+}
+
+// Tests that the total gas used by the transaction matches the expected value in the receipt,
+// and that all gas dimension components sum to the total gas consumed.
+// Scenario: CALL to a cold, funded, contract callee, no value transfer, no memory expansion.
+func TestDimTxOpCallColdNoTransferContractFundedMemUnchanged(t *testing.T) {
+	t.Parallel()
+	ctx, cancel, builder, auth, cleanup := gasDimensionTestSetup(t, false)
+	defer cancel()
+	defer cleanup()
+
+	_, caller := deployGasDimensionTestContract(t, builder, auth, gas_dimensionsgen.DeployCaller)
+	calleeAddress, _ := deployGasDimensionTestContract(t, builder, auth, gas_dimensionsgen.DeployCallee)
+
+	receipt := callOnContractWithOneArg(t, builder, auth, caller.ColdNoTransferMemUnchanged, calleeAddress)
+
+	TxOpTraceAndCheck(t, ctx, builder, receipt)
+}
+
+// Tests that the total gas used by the transaction matches the expected value in the receipt,
+// and that all gas dimension components sum to the total gas consumed.
+// Scenario: CALL to a cold, funded, contract callee, no value transfer, with memory expansion.
+func TestDimTxOpCallColdNoTransferContractFundedMemExpansion(t *testing.T) {
+	t.Parallel()
+	ctx, cancel, builder, auth, cleanup := gasDimensionTestSetup(t, false)
+	defer cancel()
+	defer cleanup()
+
+	_, caller := deployGasDimensionTestContract(t, builder, auth, gas_dimensionsgen.DeployCaller)
+	calleeAddress, _ := deployGasDimensionTestContract(t, builder, auth, gas_dimensionsgen.DeployCallee)
+
+	receipt := callOnContractWithOneArg(t, builder, auth, caller.ColdNoTransferMemExpansion, calleeAddress)
+
+	TxOpTraceAndCheck(t, ctx, builder, receipt)
+}
+
+// Tests that the total gas used by the transaction matches the expected value in the receipt,
+// and that all gas dimension components sum to the total gas consumed.
+// Scenario: CALL to a cold, virgin, no-code callee, with value transfer, no memory expansion.
+func TestDimTxOpCallColdPayingNoCodeVirginMemUnchanged(t *testing.T) {
+	t.Parallel()
+	ctx, cancel, builder, auth, cleanup := gasDimensionTestSetup(t, false)
+	defer cancel()
+	defer cleanup()
+
+	callerAddress, caller := deployGasDimensionTestContract(t, builder, auth, gas_dimensionsgen.DeployCaller)
+	// prefund the caller with some funds
+	// the TransferBalanceTo helper function does the require statements and waiting etc for us
+	_, _ = builder.L2.TransferBalanceTo(t, "Owner", callerAddress, big.NewInt(1e17), builder.L2Info)
```

Comment:
Nit. for all the TransferBalanceTo use cases
Instead of 
`_, _ = builder.L2.TransferBalanceTo(t, "Owner", callerAddress, big.NewInt(1e17), builder.L2Info)`

can skip _, _
` builder.L2.TransferBalanceTo(t, "Owner", callerAddress, big.NewInt(1e17), builder.L2Info)`